ipld/retriever.go

```diff
@@ -87,10 +89,17 @@ type retrieverSession struct {
 
 func (r *Retriever) newSession(ctx context.Context, dah *da.DataAvailabilityHeader) (*retrieverSession, error) {
 	size := len(dah.RowsRoots)
-	adder := NewNmtNodeAdder(ctx, format.NewBatch(ctx, r.dag, format.MaxSizeBatchOption(batchSize(size))))
+	adder := NewNmtNodeAdder(
+		ctx,
+		format.NewBatch(
+			ctx,
+			merkledag.NewDAGService(r.bServ),
+			format.MaxSizeBatchOption(batchSize(size)),
+		),
+	)
```

Comment:
We should add additional NodeAdder constructor from Blockservice, context and size

Reply:
Changed `NewNmtNodeAdder` ctor.